test/custom-task-ctrls/wait-task/pkg/reconciler/reconciler.go

```diff
@@ -127,9 +127,30 @@ func (c *Reconciler) ReconcileKind(ctx context.Context, r *v1alpha1.Run) kreconc
 		logger.Infof("The Custom Task Run %v timed out", r.GetName())
 		r.Status.CompletionTime = &metav1.Time{Time: c.Clock.Now()}
 		r.Status.MarkRunFailed("TimedOut", WaitTaskCancelledByRunTimeoutMsg)
+
+		// Retry if the current RetriesStatus hasn't reached the retries limit
```

Comment:
This might be a bit tricky: if we decide to change the timeout + retries behavior from v1alpha1 run -> v1beta1 customrun, this implementation will only work for one of them. It looks like this controller is responsible for reconciling v1alpha1 run. I think it would be useful to have one reconciler for testing v1alpha1 runs and one for v1beta1 customruns, but only having the reconciler for v1beta1 customruns would also be an OK decision. WDYT?